dev/tasks/exec/utils/gitignore-filter.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const fs = require( 'fs' );
+const filter = require( 'gulp-filter' );
+const gitignore = require( 'parse-gitignore' );
+const PassThrough = require( 'stream' ).PassThrough;
+
+module.exports = () => {
```

Comment:
If a module exports a function only it (the file) should be called after it. Also, the function should be named for easier debugging. This would be `filtergitignore.js` which exports `function filterGitignore()`.